test/test_saveat_solution.py

```diff
@@ -131,6 +131,27 @@ def test_saveat_solution():
     assert sol.stats["num_steps"] > 0
     assert sol.result == diffrax.RESULTS.successful
 
+    saveat = diffrax.SaveAt(steps=2)
+    sol = _integrate(saveat)
```

Comment:
We should also add tests for `SaveAt(steps=n, t1=True)` and `SaveAt(steps=n, t1=False)` when the number of steps does not divide `n`. Then check that in the first case with e.g. `n=2` and 5 steps that we store `[0 2 4 5]` and in the second case we store `[0 2 4]`